### PR TITLE
Add examples of when dynamic output filenames are important

### PR DESCRIPTION
**Pull request by mribeirodantas**
The documentation mentions that pipeline developers don't
usually have to worry about output filenames, but it's important
to mention examples of when they have to worry. This commit
works on that.

This is a response for the following feedback on Slack:
https://nextflow.slack.com/archives/C02T98A23U7/p1665098791327519

### CODE REVIEW

File: docs/process.rst [hunk 1]
```diff
@@ -632,6 +632,9 @@ with the current execution context.
   own directory, and input files are automatically staged into this directory by Nextflow.
   This behavior guarantees that input files with the same name won't overwrite each other.
 
+  An example of when you may have to deal with that is when you have many input files to a task,
+  and some of these files may have the same filename. In this case, a solution with be to use 
+  the option `stageAs`.
```
Comment (bentsherman) [inline docs/process.rst:637]: In RST, you need to use two ticks (``) on each side for code text.

**Comment by pditommaso** [comment]
Help to save polar bears adding `[ci skip]` when pushing changes for docs 😆

**Comment by mribeirodantas** [comment]
Oh, sorry. I wasn't aware of that 😬. There is a third PR that I will write today and I will do that! 🥲

**Comment by mribeirodantas** [comment]
I think it's good to go now, right?

**Comment by mribeirodantas** [comment]
LGTM, @pditommaso ?

**Review by pditommaso — APPROVED** [review]
Thanks

**Pull request merged**